frontend/src/components/ConstraintForm/ClassAccordionItem.vue

```diff
@@ -0,0 +1,55 @@
+<script>
+import { AccordionItem, AccordionList } from 'vue3-rich-accordion';
+import SubjectAccordionItem from './SubjectAccordionItem.vue';
+import { useFormStore } from '@/stores/form';
+
+export default {
+  props: {
+    index: { type: Number, default: -1 },
+  },
+  setup(props) {
+    const formStore = useFormStore();
+
+    const accordionId = `${props.index}`;
+
+    return { formStore, accordionId };
+  },
+  components: {
+    AccordionList,
+    AccordionItem,
+    SubjectAccordionItem,
+  },
+};
+</script>
+
+<template>
+  <accordion-item
+    :id="accordionId"
+    :default-opened="accordionId in formStore.accordionState ? formStore.accordionState[accordionId] : true"
```

Comment:
I believe writing it like this is easier to read:
```js
formStore?.accordionState?.[accordionId] ?? true
```

Also, it can be moved to a `computed`. I general rule of mine is to move any logic from the template to computed or methods so that the template is easier to read and that most of the logic is then contained to the script tag.